src/config/eww_config.rs

```diff
@@ -122,3 +149,153 @@ impl EwwConfig {
+            initial_variables: HashMap::new(),
+            script_vars: Vec::new(),
+        };
+
+        let merged_config = EwwConfig::merge_includes(base_config, vec![config1.unwrap(), config2.unwrap()]).unwrap();
+
+        assert_eq!(merged_config.widgets.len(), 2);
+        assert_eq!(merged_config.windows.len(), 2);
+        assert_eq!(merged_config.initial_variables.len(), 2);
+        assert_eq!(merged_config.script_vars.len(), 0);
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::config::{EwwConfig, XmlNode};
+    use std::collections::HashMap;
+
+    #[test]
+    fn test_merge_includes() {
+        let input1 = r#"
+           <eww>
+              <definitions>
+                <def name="test1">
+                  <box orientation="v">
+                    {{var1}}
+                  </box>
+                </def>
+              </definitions>
+
+              <variables>
+                <var name="var1">var1</var>
+              </variables>
+              <windows>
+                <window name="window1">
+                  <size x="100" y="200" />
+                  <pos x="100" y="200" />
+                  <widget>
+                    <test1 name="test2" />
+                  </widget>
+                </window>
+              </windows>
+            </eww>
+        "#;
+        let input2 = r#"
+            <eww>
+              <definitions>
+                <def name="test2">
+                  <box orientation="v">
+                    {{var2}}
+                  </box>
+                </def>
+              </definitions>
+              <variables>
+                <var name="var2">var2</var>
+              </variables>
+              <windows>
+                <window name="window2">
+                  <size x="100" y="200" />
+                  <pos x="100" y="200" />
+                  <widget>
+                    <test2 name="test2" />
+                  </widget>
+                </window>
+              </windows>
+            </eww>
+        "#;
+
+        let document1 = roxmltree::Document::parse(&input1).unwrap();
+        let document2 = roxmltree::Document::parse(input2).unwrap();
+        let config1 = EwwConfig::from_xml_element(XmlNode::from(document1.root_element()).as_element().unwrap().clone(), "");
+        let config2 = EwwConfig::from_xml_element(XmlNode::from(document2.root_element()).as_element().unwrap().clone(), "");
+        let base_config = EwwConfig {
+            widgets: HashMap::new(),
+            windows: HashMap::new(),
+            initial_variables: HashMap::new(),
+            script_vars: Vec::new(),
+        };
+
+        let merged_config = EwwConfig::merge_includes(base_config, vec![config1.unwrap(), config2.unwrap()]).unwrap();
+
+        assert_eq!(merged_config.widgets.len(), 2);
+        assert_eq!(merged_config.windows.len(), 2);
+        assert_eq!(merged_config.initial_variables.len(), 2);
+        assert_eq!(merged_config.script_vars.len(), 0);
+    }
+}
```

Comment:
```suggestion
```
If i'm not blind this is the same test block as the one above... no need for duplicates (in fact, probably doesn't compile even)

Reply:
That was definetly an error on my part. Idk how that ended there.  ^^